### ISSUE TITLE
Unsafe password protection

### ISSUE THREAD

**Issue by NothingHollow** [qing3962/password-protection#24]
It locks the obsidian app, however users can view the notes in a folder using other applications such as notepad. If the plugins is disabled in the settings, the notes are viewable by a user. It doesn't help much isn't it? It protects the obsidian app, not the data.

**Comment by Jabke** [comment]
Yes, i think that a trouble. Description says: "I developed this plugin for the Obsidian to prevent my girlfriend from peeking at my private notes/diaries in Obsidian.". But, dude, is your girl cant log into the folder with .md files? )). Its strange solution. Can you add the same password to folder with this files, please? Sorry for my english, but i think main idea is clear. Anyaway thank you for contribution

**Comment by NothingHollow** [comment]
> Yes, i think that a trouble. Description says: "I developed this plugin for the Obsidian to prevent my girlfriend from peeking at my private notes/diaries in Obsidian.". But, dude, is your girl cant log into the folder with .md files? )). Its strange solution. Can you add the same password to folder with this files, please? Sorry for my english, but i think main idea is clear. Anyaway thank you for contribution

Yeah, theres other add - ons that encrypts the file, however it has a bad ease of use unfortunately.

**Comment by Jabke** [comment]
> > Да, я думаю, что это проблема. В описании говорится: "Я разработал этот плагин для Obsidian, чтобы моя девушка не могла подглядывать за моими личными заметками / дневниками в Obsidian".. Но, чувак, твоя девушка не может войти в папку с файлами .md? )). Странное решение. Не могли бы вы добавить тот же пароль к папке с этими файлами, пожалуйста? Извините за мой английский, но я думаю, основная идея понятна. В любом случае спасибо за вклад.
> 
> Да, есть и другие дополнения, которые шифруют файл, однако, к сожалению, они не очень удобны в использовании.

I think by and large there is no need to encrypt all files, as some other extensions do. It is enough to set a password for the folder, although here, of course, the question arises of the cross-platform nature of such a solution. I probably need some kind of cross-platform file packaging format, something like that.zip. There is probably something more convenient, it's just a discussion on the topic...

**Comment by Jabke** [comment]
> > Yes, i think that a trouble. Description says: "I developed this plugin for the Obsidian to prevent my girlfriend from peeking at my private notes/diaries in Obsidian.". But, dude, is your girl cant log into the folder with .md files? )). Its strange solution. Can you add the same password to folder with this files, please? Sorry for my english, but i think main idea is clear. Anyaway thank you for contribution
> 
> Yeah, theres other add - ons that encrypts the file, however it has a bad ease of use unfortunately.

I think by and large there is no need to encrypt all files, as some other extensions do. It is enough to set a password for the folder, although here, of course, the question arises of the cross-platform nature of such a solution. It probably need some kind of cross-platform file packaging format, something like that.zip. There is probably something more convenient, it's just a discussion on the topic...

**Comment by qing3962** [comment]
Yes, the plugin locks the obsidian app, not the data, because it mainly apply in the mobile device, and my girl friend don't know other way to browse the files in the mobile device.

The plugin just preventing ordinary people from opening files in the Obsidian, for those familiar with mobile file system, this plugin cannot prevent them.

About："If the plugins is disabled in the settings, the notes are viewable by a user. ", If you think this is important, you can set the protection path of this plugin to "/". If you do this, when someone enter Obsidian, the plugin will ask for a password, and they will not have the opportunity to enter the settings to disable it.

" It is enough to set a password for the folder, although here, of course, the question arises of the cross-platform nature of such a solution. " -- I think this is a good idea, but it's very difficult, I currently do not have this ability.

Thank you for your discussion and suggestions.

**Comment by whoszatarash** [comment]
Greetings.

*I know this an old thread, but I wanted to note this in case that maybe, in the future, someone stumbles upon it the same way that I did.*

- - -

Something to keep in mind is that this is an _**Obsidian**_ plugin, its not meant to work outside of it. We simply cannot expect something that was made with a software infrastructure to apply to the whole OS, and the same thing goes to every other plugin out there, Obsidian or not.

There are alternatives, though. There's plenty of file-locking software available, so the best solution — albeit unpractical — is to use multiple locks in case you want to keep your stuff protected. **P.S.:** *Keep in mind that locking / encrypting a file could prevent programs tu run / open them, so be careful while testing some out*

### Things to keep in mind

- A vast majority of Obsidian users rarely check the file explorer, even less on mobile. Plugins usually aim to this majority, as the goal is to reach as many users as possible.
- The sole consideration of "the plugins (being) disabled in the settings" makes this an issue unrelated to the plugins itself and something Obsidian won't ever do (read below).
- Think calling it "unsafe" is a stretch. If the plugin promises to add password protection to notes and it does then its *useful* and *works*. Safety goes beyond development, as its up you: if you leave your devices unlocked or post your passwords online its not a softwate issue — same thing applies if someone peeks at your files.

- - -

Obsidian is [meant to be used](https://obsidian.md/about) in a way that's cloud-free, keeping your files in your devices. If this was Notion or Evernote it might've worked that way, but that's not the case.

**TL;DR:** If the plugin was made for Obsidian, it won't work outside of it.

***Thanks for reading and have a nice week! :)***
